stagiaires/oshi/exercices/JS-EX-03/initiation.js

```diff
@@ -0,0 +1,290 @@
+  document.getElementsByClassName("data-controls")[0].appendChild(new_input);
+  index_value++;
+})
+
+function permutte(sens) {
+  const tab = exo[11].c1;
+  let tab_tmp = [];
+
+  if (sens === 1) {
+    tab_tmp = Array.from(tab).slice(1).map(input => input.value);
+    tab_tmp.push(tab[0].value);
+  } else {
+    tab_tmp = Array.from(tab).slice(0, tab.length - 1).map(input => input.value);
+    tab_tmp.unshift(tab[tab.length - 1].value);
+  }
+
+  tab.forEach((element, c) => {
+    element.value = tab_tmp[c];
+  })
+}
+
+exo[12].droite.addEventListener('click', () => {
+  Array.from(exo[12].s1.options).forEach(option => {
+    if (option.selected) {
+      exo[12].s2.appendChild(option);
+    }
+  });
+})
+
+exo[12].gauche.addEventListener('click', () => {
+  Array.from(exo[12].s2.options).forEach(option => {
+    if (option.selected) {
+      exo[12].s1.appendChild(option);
+    }
+  });
+})
+
+exo[13].c02.disabled = true;
+exo[13].c03.disabled = true;
+exo[13].addEventListener('keyup', () => {
+  if(exo[13].c01.value.length === 5) {
+    exo[13].c02.disabled = false;
+    exo[13].c02.focus();
+    if(exo[13].c02.value.length === 5) {
+      exo[13].c03.disabled = false;
+      exo[13].c03.focus();
+    } else {
+      exo[13].c03.disabled = true;
+      exo[13].c03.value = null;
+    }
+  } else {
+    exo[13].c02.disabled = true;
+    exo[13].c02.value = null;
+    exo[13].c03.disabled = true;
+    exo[13].c03.value = null;
+  }
+})
+
+exo[14].cb1.addEventListener('click', () => {
+  if (exo[14].cb1.checked) {
+    exo[14].ch1.style.display = "none";
+  } else {
+    exo[14].ch1.style.display = "block";
+  }
+})
+
+const actual_style = exo[15].ch1.style;
+actual_style.width="170px";
+actual_style.height="16px";
+exo[15].ch1.addEventListener('focus', () => {
+  actual_style.width = Number(parseInt(actual_style.width)+100)+"px";
+  actual_style.height = Number(parseInt(actual_style.height)+100)+"px";
+})
+exo[15].ch1.addEventListener('blur', () => {
+  actual_style.width = "170px";
+  actual_style.height = "16px";
+})
+
+const exo_carres_1 = Array.from(document.getElementsByClassName("1_exo_carre carre"));
+exo_carres_1.forEach((carre) => {
+  carre.addEventListener("click", () => {
+    exo_carres_1.map((allCarre) => allCarre.classList.remove("checked"));
+    carre.classList.add("checked");
+  });
+});
+
+const exo_carres_2 = Array.from(document.getElementsByClassName("2_exo_carre carre"));
+const exo_classe_2 = document.getElementById("out").classList;
+let last_class = "blanc";
+exo_carres_2.forEach((carre) => {
+  carre.addEventListener("click", () => {
+    exo_carres_2.map((allCarre) => allCarre.classList.remove("checked"));
+    carre.classList.add("checked");
+    exo_classe_2.replace(last_class, carre.getAttribute("data-color"));
+    last_class = carre.getAttribute("data-color");
+  });
+});
+const imgs_url = 'https://pixabay.com/api/?key=15935321-ecb5695630be7347dafccf251&q=developement&per_page=200';
```

Comment:
L'utilisateur peut faire soit le drag and drop, soit le click.  Donc les deux doivent fonctionner, or si je clique sur le vert  l'image de droite doit devenir verte

Reply:
> L'utilisateur peut faire soit le drag and drop, soit le click. Donc les deux doivent fonctionner, or si je clique sur le vert l'image de droite doit devenir verte

Je remet l'énoncé : Soit on clique sur l'image à droite pour lui donner le src de la dernière image de gauche sur la quelle on a cliquer

Si tu clique sur l'image vert puis sur l'image de droit elle deviendra bien verte ;)